pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,239 @@
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// For each revision, check that the configmap for that revision have
+	// correct argument values or not
+	revisionsHaveSynced := true
+	for _, revision := range uniqueRevisions {
+		configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+		configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		matches := true
+		for arg, val := range desiredControllerManagerArgumentVals {
+			match, err := configMatchesControllerManagerArgument(configMap, arg, val)
+			if err != nil {
+				return err
+			}
+			matches = matches && match
+		}
+		if !matches {
+			revisionsHaveSynced = false
+			break
+		}
+	}
+
+	conditions := []metav1.Condition{
+		{
+			Type:   apiconfigv1.KubeControllerManagerComplete,
+			Status: metav1.ConditionUnknown,
+			Reason: reasonWorkerLatencyProfileUpdated,
+		}, {
+			Type:   apiconfigv1.KubeAPIServerProgressing,
+			Status: metav1.ConditionUnknown,
+			Reason: reasonWorkerLatencyProfileUpdated,
+		}, {
+			Type:   apiconfigv1.KubeAPIServerDegraded,
+			Status: metav1.ConditionUnknown,
+		},
+	}
+	if revisionsHaveSynced {
+		// APIServer has Completed
+		conditions[0].Status = metav1.ConditionTrue
+		conditions[0].Message = "all kube-controller-manager(s) have updated latency profile"
+
+		// APIServer is not Progressing
+		conditions[1].Status = metav1.ConditionFalse
+
+		// APIServer is not Degraded
+		conditions[2].Status = metav1.ConditionFalse
+
+	} else {
+		// APIServer has not Completed
+		conditions[0].Status = metav1.ConditionFalse
+
+		// APIServer is Progressing
+		conditions[1].Status = metav1.ConditionTrue
+		conditions[1].Message = "kube-controller-manager(s) are updating latency profile"
+
+		// APIServer is not Degraded
+		conditions[2].Status = metav1.ConditionFalse
+	}
+	_, err = c.updateConfigNodeStatus(conditions...)
+	return err
+}
+
+// configHasControllerManagerArgument checks if the specified config map containing kcm node config
+// contains the specified argument and value in observedconfig.extendedarguments field
+func configMatchesControllerManagerArgument(configMap *corev1.ConfigMap, arg, value string) (bool, error) {
+	configData, ok := configMap.Data[controllerManagerConfigMapKey]
+	if !ok {
+		return false, fmt.Errorf("could not find %s in %s config map from %s namespace", controllerManagerConfigMapKey, configMap.Name, configMap.Namespace)
+	}
+	var kubeControllerManagerConfig controlplanev1.KubeControllerManagerConfig
+	if err := json.Unmarshal([]byte(configData), &kubeControllerManagerConfig); err != nil {
```

Comment:
if we have the whole desiredControllerManagerArgumentVals map here we don't have to do unmarshal multiple times

Reply:
Updated tit accordingly with function signature as `func configMatchesControllerManagerArguments(configMap *corev1.ConfigMap, argValMap map[string]string) (bool, error)`.